src/tidy.zig

```diff
@@ -62,30 +80,104 @@ test "tidy" {
+    // - use fixme for issues to be fixed in the same pull request,
+    // - use todo as general-purpose long-term remainders without enforcement.
+    if (std.mem.indexOf(u8, source, "FIX" ++ "ME") != null) {
+        return "FIX" ++ "ME comments must be addressed before getting to main";
+    }
+
+    return null;
 }
 
-fn tidy_long_line(file: SourceFile) !void {
-    if (std.mem.endsWith(u8, file.path, "low_level_hash_vectors.zig")) return;
-    const long_line = try find_long_line(file.text);
-    if (long_line) |line_index| {
-        std.debug.print(
-            "{s}:{d} error: line exceeds 100 columns\n",
-            .{ file.path, line_index + 1 },
-        );
-        return error.LineTooLong;
+fn tidy_long_line(file: SourceFile) !?u32 {
+    if (std.mem.endsWith(u8, file.path, "low_level_hash_vectors.zig")) return null;
+    var line_iterator = mem.split(u8, file.text, "\n");
+    var line_index: u32 = 0;
+    while (line_iterator.next()) |line| : (line_index += 1) {
+        const line_length = try std.unicode.utf8CountCodepoints(line);
+        if (line_length > 100) {
+            if (has_link(line)) continue;
+
+            // Journal recovery table
+            if (std.mem.indexOf(u8, line, "Case.init(") != null) continue;
+
+            // For multiline strings, we care that the _result_ fits 100 characters,
+            // but we don't mind indentation in the source.
+            if (parse_multiline_string(line)) |string_value| {
+                const string_value_length = try std.unicode.utf8CountCodepoints(string_value);
+                if (string_value_length <= 100) continue;
+
+                if (std.mem.startsWith(u8, string_value, " account A") or
+                    std.mem.startsWith(u8, string_value, " transfer T") or
+                    std.mem.startsWith(u8, string_value, " transfer   "))
+                {
+                    // Table tests from state_machine.zig. They are intentionally wide.
+                    continue;
+                }
+
+                // vsr.zig's Checkpoint ops diagram.
+                if (std.mem.startsWith(u8, string_value, "OPS: ")) continue;
+            }
+
+            return line_index;
+        }
     }
+    return null;
+}
+
+fn tidy_control_characters(file: SourceFile) ?u8 {
+    const binary_file_extensions: []const []const u8 = &.{ ".ico", ".png" };
+    for (binary_file_extensions) |extension| {
+        if (std.mem.endsWith(u8, file.path, extension)) return null;
+    }
+
+    if (mem.indexOfScalar(u8, file.text, '\r') != null) {
+        if (std.mem.endsWith(u8, file.path, ".bat")) return null;
+        return '\r';
+    }
+
+    // Learning the best from UNIX, Visual Studio, like make, insists on tabs.
+    if (std.mem.endsWith(u8, file.path, ".sln")) return null;
+    // Go code uses tabs.
+    if (std.mem.endsWith(u8, file.path, ".go") or
+        (std.mem.endsWith(u8, file.path, ".md") and mem.indexOf(u8, file.text, "```go") != null))
```

Comment:
Since all .md with GO code should be from auto-generated client readmes, WDYT of replacing tabs directly at `client_readmes.read_section()`?

Reply:
I assume we actually want to have tabs, to make the code copy-pastable?

Reply:
> to make the code copy-pastable?

Not a big deal I think.

Tabs are just idiomatic for Go and enforced by `go fmt`. The code might still be valid, differently from Zig, which rejects tabs at the compiler level! :laughing: 

Anyway, it's good to have 100% idiomatic samples there.